packages/fxa-auth-server/fxa-oauth-server/lib/db/memory.js

```diff
@@ -309,6 +309,63 @@ MemoryStore.prototype = {
     return helpers.aggregateActiveClients(activeClientTokens);
   },
 
+  /**
+   * Get all access tokens for a given user.
+   * @param {String} uid User ID as hex
+   * @returns {Promise}
+   */
+  getAccessTokensByUid: async function getAccessTokensByUid(uid) {
+    if (! uid) {
+      return P.reject(new Error('Uid is required'));
+    }
+    const accessTokens = [];
+    Object.keys(this.tokens).forEach(id => {
+      const token = this.tokens[id];
+      if (token.userId.toString('hex') === uid) {
+        var clientIdHex = unbuf(token.clientId);
+        var client = this.clients[clientIdHex];
+        accessTokens.push({
+          accessTokenId: buf(id),
+          clientId: token.clientId,
+          createdAt: token.createdAt,
+          clientName: client.name,
+          clientCanGrant: client.canGrant,
+          scope: this.tokens[id].scope
+        });
+      }
+    });
+    return accessTokens;
+  },
+
+  /**
+   * Get all refresh tokens for a given user.
+   * @param {String} uid User ID as hex
+   * @returns {Promise}
+   */
+  getRefreshTokensByUid: async function getRefreshTokensByUid(uid) {
+    if (! uid) {
+      return P.reject(new Error('Uid is required'));
```

Comment:
nit: This feels like a bit of a syntactic mismatch to return a `Promise.reject` instead of throw'ing in an async function.